core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -220,6 +227,11 @@ public IndexNameExpressionResolver(Settings settings) {
         return concreteIndices.toArray(new Index[concreteIndices.size()]);
     }
 
+    private static IllegalArgumentException aliasesNotSupportedException(String expression) {
+        return new IllegalArgumentException("The provided expression [" + expression + "] matches an " +
+                "alias, specify the corresponding concrete indices instead.");
```

Comment:
but it can be more than one index as an alias can point to multiple indices. Would you change that anyway?